internal/telemetry/collector.go

```diff
@@ -110,26 +97,47 @@ func (c *Collector) Collect(ctx context.Context) {
 	if err != nil {
 		glog.Errorf("Error exporting telemetry data: %v", err)
 	}
-	glog.V(3).Infof("Exported telemetry data: %x", traceData)
+	glog.V(3).Infof("Exported telemetry data: %+v", traceData)
 }
 
-// Start starts running NIC Telemetry Collector.
-func (c *Collector) Start(ctx context.Context) {
-	wait.JitterUntilWithContext(ctx, c.Collect, c.Period, 0.1, true)
+// BuildReport takes context and builds report from gathered telemetry data.
+func (c *Collector) BuildReport(ctx context.Context) (TraceData, error) {
+	dt := TraceData{}
+	var err error
+
+	for _, namespace := range c.Config.Namespaces {
+		dt.VirtualServers += c.GetVirtualServerCount(ctx, namespace)
+		dt.TransportServers += c.GetTransportServerCount(ctx, namespace)
+	}
+
+	return dt, err
 }
 
-// GetVSCount returns number of VirtualServers in watched namespaces.
-//
-// Note: this is a placeholder function.
-func (c *Collector) GetVSCount() int {
-	// Placeholder function
-	return 0
+// SetConfig is used to overwrite the existing config for the Collector.
+func (c *Collector) SetConfig(cfg CollectorConfig) {
+	c.Config = cfg
 }
 
-// GetTSCount returns number of TransportServers in watched namespaces.
-//
-// Note: this is a placeholder function.
-func (c *Collector) GetTSCount() int {
-	// Placeholder function
-	return 0
+// GetVirtualServerCount returns the number of VirtualServers in a namespace.
+func (c *Collector) GetVirtualServerCount(ctx context.Context, namespace string) int {
+	count, err := c.Config.CustomK8sClientReader.
+		K8sV1().
+		VirtualServers(namespace).
+		List(ctx, meta_v1.ListOptions{})
+	if err != nil {
+		glog.Errorf("unable to list VirtualServers in namespace %s Error: %x", namespace, err)
```

Comment:
if there is an error here it may be worth to consider returning 0

Reply:
That should work. If we're not able to list a resource in a certain namespace, then it should just add `0`